cmd/tests/tests_test.go

```diff
@@ -0,0 +1,53 @@
+// Package tests contains integration tests for multiple packages.
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"sync/atomic"
+	"testing"
+)
+
+type blockingTransport struct {
```

Comment:
hmm currently this thing and `TestMain` are duplicated in `cmd/root_test.go` 

It is probably a good thing to have them in both places, though maybe they can be exported from this package and imported in `cmd/root_test.go`? :thinking:

Reply:
Right, I overlooked this.

Instead of exporting only `blockingTransport` and duplicating `TestMain` in both packages, I added a new function `tests.Main`, and called it from each package's `TestMain()`. See 895a657a.

This way we don't have to redefine all the hosts logic and goleak check, which is more significant.